### PR TITLE
Create dummy payment and add to callback

### PR DESCRIPTION
**Pull request by tomxelliott**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java [hunk 1]
```diff
@@ -81,6 +87,15 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             details.getId()
         );
 
+        log.info("Setting dummy payment to mock payment process");
+        if (caseData.getPayments() == null || caseData.getPayments().isEmpty()) {
+            List<ListValue<Payment>> payments = new ArrayList<>();
+            payments.add(solicitorSubmitPetitionService.getDummyPayment(orderSummary));
+            caseData.setPayments(payments);
+        } else {
+            caseData.getPayments().add(solicitorSubmitPetitionService.getDummyPayment(orderSummary));
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:96]: so if there is a payment in the list we will add another one?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitPetitionService.java [hunk 2]
```diff
@@ -57,4 +60,22 @@ public OrderSummary getOrderSummary() {
             )
             .build();
     }
+
+    public ListValue<Payment> getDummyPayment(OrderSummary orderSummary) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitPetitionService.java:64]: it kind of feels that this thing should return a Payment and then the callback handler should add the ListValue wrapper. I know it's only temporary but in theory the Payment might not always be used in a list

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java (hunk 1 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:96]: Yeah, from what I can gather from other use services use cases, if a payment were to fail for one reason or another we would want to store it in our payments array for future records - so any subsequent payments would be added to the payments array and we have a record of attempted and successful payments on the case.

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitPetitionService.java (hunk 2 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitPetitionService.java:64]: 👍 - was in two minds about the return type for this one - I agree that returning a Payment makes the most sense

File: src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+@Data
+@Builder
+public class Payment {
+
+    @CCD(
+        label = "Payment Date",
+        access = {DefaultAccess.class}
+    )
+    @JsonProperty("PaymentDate")
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java:22]: weren't we going to remove all these?

File: src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import java.time.LocalDate;
+
+@Data
+@Builder
+public class Payment {
+
+    @CCD(
+        label = "Payment Date",
+        access = {DefaultAccess.class}
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java:20]: Do fields on a complex type have access control or is it only the complex type as a whole?

File: src/main/java/uk/gov/hmcts/divorce/common/model/CaseData.java [hunk 5]
```diff
@@ -724,6 +725,14 @@
     )
     private Set<FinancialOrderFor> financialOrderFor;
 
+    @CCD(
+        label = "Payments",
+        typeOverride = Collection,
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/common/model/CaseData.java:730]: were the overrides still needed?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java [hunk 6]
```diff
@@ -80,6 +87,23 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
+            List<ListValue<Payment>> payments = new ArrayList<>();
+            payments.add(
+                ListValue
+                    .<Payment>builder()
+                    .value(solicitorSubmitPetitionService.getDummyPayment(orderSummary))
+                    .build());
+            caseData.setPayments(payments);
+        } else {
+            caseData.getPayments().add(
+                ListValue
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:101]: isn't there a `new ListValue(item)`? We might want a static helper method in future: `listValue(item)`

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java [hunk 7]
```diff
@@ -110,6 +134,20 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             .build();
     }
 
+    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
+                                               CaseDetails<CaseData, State> beforeDetails) {
+        final CaseData caseData = details.getData();
+        final PaymentStatus paymentStatus = caseData.getPayments().get(caseData.getPayments().size() - 1).getValue().getPaymentStatus();
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java:140]: Feels like we might need something a bit more robust here? Sum the total of successful payments and make sure it is the same value as the Fee?

File: src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java (hunk 3 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/payment/model/Payment.java:22]: Yep, completely forgot to remove these. well spotted!

**Review by linusnorton — APPROVED** [review]

**Pull request merged**